src/gpu_sysman.c

```diff
@@ -900,54 +901,54 @@ static bool gpu_ras(gpu_device_t *gpu) {
         // https://spec.oneapi.io/level-zero/latest/sysman/PROG.html#querying-ras-errors
       case ZES_RAS_ERROR_CAT_RESET:
         help = "Total count of HW accelerator resets attempted by the driver";
-        catname = METRIC_PREFIX "resets_total";
+        catname = METRIC_PREFIX "errors.resets";
```

Comment:
> Can the errors.* metrics be mapped to hw.errors, with the `hw.error.type` label indicating the type of error?

Errors part of spec looks really annoying, and seems to require something stupid like this (instead of what you proposed):
`hw.errors{hw.type="gpu",hw.error.type="uncorrectable",hw.error.gpu.type="reset"}`

Whereas I'd rather use something (clearly) more readable, and more consistent with the other GPU metrics in the spec:
`hw.gpu.errors.resets{hw.error.type="uncorrectable"}`

Comments?

Reply:
Btw. spec already has `hw.gpu.power`: https://opentelemetry.io/docs/specs/semconv/system/hardware-metrics/#hwgpu---gpu-metrics

and `hw.power{hw.type="gpu"}`: https://opentelemetry.io/docs/specs/semconv/system/hardware-metrics/#hw---common-hardware-metrics

So same for errors seems natural.

Reply:
My guess is that the idea is to use `hw.errors` (and `hw.status`) for all hardware, so that you can combine them in one graph, showing you any hardware error that occurred on the system. The `hw.type` label allows you to aggregate by hardware type, e.g. filter for GPU errors or sum up GPU errors.

Judging from other sections, data points are supposed to have one other label, `hw.error.type`. The values ("corrected", "uncorrected") are only examples. This is where the differentiation between different errors should happen. For the gpu_sysman plugin, we should set this to "reset", "programming", "driver", …

The duplication between `hw.power`, `hw.gpu.power` is weird. I'd probably prefer `hw.power` so users can aggregate the power consumption of different components.

Reply:
> > The values ("corrected", "uncorrected") are only examples.
> 
> No, they're something that need to be in GPU metrics / were already provided by Sysman metrics `type` label (which needs now `hw.error.` prefix). I.e. the actual error names like `reset` etc, would need to use some other label, maybe `hw.error.name`?

Other devices suggest other values for `hw.error.type`, e.g. network interfaces suggest "zero_buffer_credit, crc, etc." and disks suggest "bad_sector, write, etc.". For me it is quite clear that this is meant as "whatever is relevant for the device goes here" kind of placeholders.

If all GPU errors come in corrected and uncorrected variants, I'd actually suggest to introduce a new label, `hw.error.correction` and set this to either "corrected" or "uncorrected" instead.

> As to power aggregation, there's separate metric category for whole host power

That makes sense to me. The host power is an aggregate of sorts, so it makes sense to define a separate metric for it.

> Weirdly it does not list actual power usage, just limit & utilization.

That's really weird.

>  And worse, it does not require label for which component's PSU the data is for (whole host, GPU, some other component?), WTF?

They do recommend a `serial_number` attribute, which can use used to differentiate different power supplies. I think if there are dedicated power supplies for mainboard/cpu on one hand and GPU(s) on the other hand, then it would make perfect sense to me to tag them accordingly.

Reply:
> If all GPU errors come in corrected and uncorrected variants, I'd actually suggest to introduce a new label, hw.error.correction and set this to either "corrected" or "uncorrected" instead.

Some of them are not correctable (e.g. resets), but most are: https://spec.oneapi.io/level-zero/latest/sysman/PROG.html#querying-ras-errors

Both above oneAPI Level Zero spec and OpenTelemetry spec state that to be error "type", so I would use that.

As to whether the error is GPU reset, in HW programming, in cache or other location, the term used by the L0 spec for that is "category": https://spec.oneapi.io/level-zero/latest/sysman/api.html#zes-ras-error-type-t

So, maybe the label could be "hw.error.category"?

> They do recommend a serial_number attribute, which can use used to differentiate different power supplies.

I do not see how user is supposed to determinate one "random" (serial) number to mean GPU, and another "random" (serial) number to mean PSU. :-)